src/core/thread/mle_router.cpp

```diff
@@ -2620,17 +2612,17 @@ ThreadError MleRouter::SendChildIdResponse(Child *aChild)
         while (FindChild(mNextChildId) != NULL);
 
         // allocate Child ID
-        aChild->mValid.mRloc16 = mNetif.GetMac().GetShortAddress() | mNextChildId;
+        aChild->SetRloc16(mNetif.GetMac().GetShortAddress() | mNextChildId);
     }
 
-    SuccessOrExit(error = AppendAddress16(*message, aChild->mValid.mRloc16));
+    SuccessOrExit(error = AppendAddress16(*message, aChild->GetRloc16()));
 
-    for (uint8_t i = 0; i < sizeof(aChild->mRequestTlvs); i++)
+    for (uint8_t i = 0; i < Child::kMaxRequestTlvs; i++)
     {
-        switch (aChild->mRequestTlvs[i])
+        switch (aChild->GetRequestTlv(i))
         {
         case Tlv::kNetworkData:
-            SuccessOrExit(error = AppendNetworkData(*message, (aChild->mMode & ModeTlv::kModeFullNetworkData) == 0));
+            SuccessOrExit(error = AppendNetworkData(*message, !aChild->GetFullNetworkData()));
```

Comment:
Clarity check: a naive reading of `GetFullNetworkData` would be that the method returns a large blob of data, not a boolean.  Would `HasFullNetworkData` be better?